File: daily/2023.11/2656.maximum-sum-with-exactly-k-elements.py
'''
Author: TDK 793065367@qq.com
Date: 2023-11-15 00:00:01
LastEditors: TDK 793065367@qq.com
LastEditTime: 2023-11-15 10:13:40
FilePath: /My-Leetcode/daily/2023.11/2656.maximum-sum-with-exactly-k-elements.py
Description: 这是默认设置,请设置`customMade`, 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
'''
"""
日期: 2023-11-15
题目: K 个元素的最大和
链接: https://leetcode-cn.com/problems/maximum-sum-with-exactly-k-elements/
"""

from typing import *
from collections import *
class Solution:
    def maximizeSum(self, nums: List[int], k: int) -> int:
        return (2 * max(nums) + k - 1) * k // 2

File: LeetCodeGo/2656.maximum-sum-with-exactly-k-elements/2656.maximum-sum-with-exactly-k-elements.go
/* INTRODUCTION
 *日期: 2023-11-16
 *题目: K 个元素的最大和
 *链接: https://leetcode-cn.com/problems/maximum-sum-with-exactly-k-elements/
 */

package main

import "fmt"

// --CODE BEGIN--
func maximizeSum(nums []int, k int) int {
	maxNum := 0
	for _, num := range nums {
		if num > maxNum {
			maxNum = num
		}
	}
	return (2*maxNum + k - 1) * k / 2
}

//--CODE END--

/*--TESTCASE BEGIN--
 *TestCase 0: [1,2,3,4,5],3
 *TestCase 1: [5,5,5],2
 *--TESTCASE END--
 */
func main() {
	fmt.Print(maximizeSum([]int{1, 2, 3, 4, 5}, 3))
}
